From 034a8f18d255052c9598a481e0f2b44292c6c8d7 Mon Sep 17 00:00:00 2001
From: "Rywacki, KamilX" <kamilx.rywacki@intel.com>
Date: Tue, 26 Feb 2019 19:14:26 +0100
Subject: [PATCH] PONRTSYS-3696: Wrong arbitration mode is set for scheduler

- convert DP arbi enum to PPv4 arbi enum in dp_link_set function
---
 drivers/net/ethernet/lantiq/datapath/gswip31/datapath_ppv4_api.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_ppv4_api.c b/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_ppv4_api.c
index 9c6c46772107..e329ea9722b4 100644
--- a/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_ppv4_api.c
+++ b/drivers/net/ethernet/lantiq/datapath/gswip31/datapath_ppv4_api.c
@@ -2189,7 +2189,7 @@ static int dp_link_set(struct dp_node_link *info, int parent_node, int flag)
 		qos_sched_conf_set_default(sched_cfg);
 		sched_cfg->sched_child_prop.parent = parent_node;
 		sched_cfg->sched_child_prop.priority = info->prio_wfq;
-		sched_cfg->sched_parent_prop.arbitration = info->arbi;
+		sched_cfg->sched_parent_prop.arbitration = arbi_dp2pp(info->arbi);
 		node_id = info->node_id.sch_id;
 
 		DP_DEBUG(DP_DBG_FLAG_QOS,
